src/plugins/intel_gpu/tests/test_cases/reverse_gpu_test.cpp

```diff
@@ -0,0 +1,355 @@
+    std::vector<T> expected_out;
+};
+
+template <typename T, reverse_mode mode>
+struct reverse_gpu_test : public ::testing::TestWithParam<ReverseParams<T, mode>> {
+public:
+    void test() {
+        auto data_type = type_to_data_type<T>::value;
+        ReverseParams<T, mode> params = testing::TestWithParam<ReverseParams<T, mode>>::GetParam();
+        auto& engine = get_test_engine();
+        tensor t;
+
+        auto reverse_input = engine.allocate_memory({data_type, params.input_format, params.input_tensor});
+        auto reverse_axes = engine.allocate_memory(
+            {ReverseModeTraits<mode>::data_type, format::bfyx, tensor(batch(1), feature(params.axis.size()))});
+
+        set_values(reverse_input, params.input);
+        set_values(reverse_axes, params.axis);
+
+        const std::string reverse_id = "reverse";
+        const std::string reverse_input_id = "reverse_input";
+        const std::string axes_id = "reverse_axes";
+        topology topology;
+        topology.add(input_layout(reverse_input_id, reverse_input->get_layout()));
+        topology.add(input_layout(axes_id, reverse_axes->get_layout()));
+
+        topology.add(reverse(reverse_id, reverse_input_id, axes_id, mode));
+
+        network network(engine, topology);
+
+        network.set_input_data(reverse_input_id, reverse_input);
+        network.set_input_data(axes_id, reverse_axes);
+
+        auto result = network.execute();
+        auto out_mem = result.at(reverse_id).get_memory();
+        cldnn::mem_lock<T> out_ptr(out_mem, get_test_stream());
+
+        ASSERT_EQ(params.expected_out.size(), out_ptr.size());
+        for (size_t i = 0; i < params.expected_out.size(); ++i) {
+            EXPECT_NEAR(params.expected_out[i], out_ptr[i], 0.0001) << "at i = " << i;
+        }
+    }
+};
+
+struct PrintToStringParamName {
+    template <class T, reverse_mode mode>
+    std::string operator()(const testing::TestParamInfo<ReverseParams<T, mode>>& param) {
+        std::stringstream buf;
+        buf << "input tensor " << param.param.input_tensor.to_string();
+        buf << " axes {";
+        for (auto val : param.param.axis) {
+            buf << val << ",";
+        }
+        buf << "} format" << param.param.input_format.to_string();
+        return buf.str();
+    }
+};
+
+using reverse_gpu_test_int32_mask = reverse_gpu_test<int32_t, reverse_mode::mask>;
+using reverse_gpu_test_int32_index = reverse_gpu_test<int32_t, reverse_mode::index>;
+using reverse_gpu_test_int64_mask = reverse_gpu_test<int64_t, reverse_mode::mask>;
+using reverse_gpu_test_int64_index = reverse_gpu_test<int64_t, reverse_mode::index>;
+using reverse_gpu_test_float_mask = reverse_gpu_test<float, reverse_mode::mask>;
+using reverse_gpu_test_float_index = reverse_gpu_test<float, reverse_mode::index>;
+using reverse_gpu_test_int8_mask = reverse_gpu_test<int8_t, reverse_mode::mask>;
+using reverse_gpu_test_int8_index = reverse_gpu_test<int8_t, reverse_mode::index>;
```

Comment:
Could you please add test case for uint8 data type as it is supported in the primitive?

Reply:
done